language/tools/test_generation/src/bytecode_generator.rs

```diff
@@ -290,7 +292,10 @@ impl BytecodeGenerator {
         summaries::instruction_summary(instruction)
             .effects
             .iter()
-            .fold(state, |acc, effect| effect(&acc))
+            .fold(state, |acc, effect| {
+                effect(&acc)
+                    .unwrap_or_else(|err| panic!("Error applying instruction effect: {}", err))
```

Comment:
panic does not seem like the best way to deal with this.

Reply:
Added a TODO comment in next PR.